src/fuzzer.rs

```diff
@@ -1,229 +1,135 @@
+    coverage_clients::{
+        CoverageClient,
+        endpoint::{EndpointCoverageClient, setup_endpoint_coverage},
+        setup_line_coverage, validate_instrumentation,
+    },
     executor::SequenceExecutor,
+    initial_corpus::minimize_corpus,
     input::OpenApiInput,
-    monitors::CoverageMonitor,
+    monitors::construct_event_mgr,
+    openapi::parse_api_spec,
     openapi_mutator::havoc_mutations_openapi,
-    state::OpenApiFuzzerState,
+    reporting::{generate_coverage_reports, generate_report_path},
+    state::{construct_state_uninit, init_state},
+    types::{CombinedMapObserverType, ObserversTupleType, OpenApiFuzzerStateType, SchedulerType},
 };
 
 /// Main fuzzer function.
 ///
 /// Sets up the various nuts and bolts required by LibAFL and runs the fuzzer until the configured
 /// timeout is reached, or until a (ctrl-c) interrupt is caught.
 pub fn fuzz() -> Result<()> {
+    // Preparatory stuff
     let config = &Configuration::get().map_err(anyhow::Error::msg)?;
     crate::setup_logging(config);
     let report_path = config.report.then(generate_report_path);
+    let api = parse_api_spec(config)?;
 
-    // Override the `servers` field in the OpenAPI specification if a server override
-    // was specified on the CLI.
-    let mut api = crate::openapi::get_api_spec(config.openapi_spec.as_ref().unwrap())?;
-    if let Some(server_override) = &config.target {
-        api.servers = vec![Server {
-            url: server_override.as_str().trim_end_matches('/').to_string(),
-            description: None,
-            variables: None,
-            extensions: IndexMap::new(),
-        }];
-    }
+    let initial_corpus = crate::initial_corpus::initialize_corpus(
+        &api,
+        config.initial_corpus.as_deref(),
+        &report_path.as_deref(),
+    );
+    let mut state_uninit = construct_state_uninit(&api, initial_corpus)?;
 
-    // The Monitor trait define how the fuzzer stats are reported to the user
-    let mon = CoverageMonitor::new(|s| info!("{s}"));
+    let mutator_openapi = HavocScheduledMutator::new(havoc_mutations_openapi());
 
-    // The event manager handle the various events generated during the fuzzing loop
-    // such as the notification of the addition of a new item to the corpus
-    let mut mgr = SimpleEventManager::new(mon);
+    // Event manager
+    let mut mgr = construct_event_mgr();
 
-    // Set up endpoint coverage
+    // Observers & feedback initialization
     let (mut endpoint_coverage_client, endpoint_coverage_observer, endpoint_coverage_feedback) =
-        setup_endpoint_coverage(*api.clone())?;
-
+        setup_endpoint_coverage(api.clone())?;
     let (mut code_coverage_client, code_coverage_observer, code_coverage_feedback) =
         setup_line_coverage(config, &report_path)?;
+    let (time_observer, time_feedback) = {
+        let observer = TimeObserver::new("time");
+        let feedback = TimeFeedback::new(&observer);
+        (observer, feedback)
+    };
```

Comment:
Seems esoteric to me. Doesn't it just work like this?
```suggestion
    let time_observer = TimeObserver::new("time");
    let time_feedback = TimeFeedback::new(&time_observer);
```

Reply:
Sadly it doesn't. Your suggestion leads to ownership complaints.

Reply:
you forgot to do the [crab dance](https://www.youtube.com/watch?v=LDU_Txk06tM) to appease Cargo